tests/Unit/Elliptic/Systems/Xcts/Events/Test_ObserveAdmIntegrals.cpp

```diff
@@ -6,156 +6,245 @@
+        make_with_value<tnsr::I<DataVector, 3, Frame::Inertial>>(
+            inertial_coords, 0.0);
+    for (int i = 0; i < 3; i++) {
+      for (int j = 0; j < 3; j++) {
+        barred_coords.get(i) +=
+            boost_matrix.get(i + 1, j + 1) * inertial_coords.get(j);
+      }
+    }
+
+    // Get barred spacetime variables.
+    const auto barred_spacetime_vars = solution.variables(
+        barred_coords,
+        tmpl::list<gr::Tags::Lapse<DataVector>,
+                   gr::Tags::Shift<DataVector, 3, Frame::Inertial>,
+                   gr::Tags::SpatialMetric<DataVector, 3, Frame::Inertial>>{});
+    const auto& barred_lapse =
+        get<gr::Tags::Lapse<DataVector>>(barred_spacetime_vars);
+    const auto& barred_shift =
+        get<gr::Tags::Shift<DataVector, 3, Frame::Inertial>>(
+            barred_spacetime_vars);
+    const auto& barred_spatial_metric =
         get<gr::Tags::SpatialMetric<DataVector, 3, Frame::Inertial>>(
-            background_fields);
+            barred_spacetime_vars);
+
+    // Construct barred spacetime metric.
+    tnsr::aa<DataVector, 3, Frame::Inertial> barred_spacetime_metric;
+    gr::spacetime_metric(make_not_null(&barred_spacetime_metric),
+                         barred_lapse, barred_shift, barred_spatial_metric);
+
+    // Transform spacetime metric into the inertial frame.
+    auto spacetime_metric =
+        make_with_value<tnsr::aa<DataVector, 3, Frame::Inertial>>(
+            inertial_coords, 0.0);
+    for (int a = 0; a < 4; a++) {
+      for (int b = 0; b <= a; b++) {
+        for (int c = 0; c < 4; c++) {
+          for (int d = 0; d < 4; d++) {
+            spacetime_metric.get(a, b) += boost_matrix.get(c, a) *
+                                          boost_matrix.get(d, b) *
+                                          barred_spacetime_metric.get(c, d);
+          }
+        }
+      }
+    }
+
+    // Do the 3+1 decomposition.
+    const auto spatial_metric = gr::spatial_metric(spacetime_metric);
     const auto& inv_spatial_metric =
-        get<gr::Tags::InverseSpatialMetric<DataVector, 3, Frame::Inertial>>(
-            background_fields);
-    const auto& extrinsic_curvature =
-        get<gr::Tags::ExtrinsicCurvature<DataVector, 3, Frame::Inertial>>(
-            background_fields);
-    const auto& trace_extrinsic_curvature =
-        get<gr::Tags::TraceExtrinsicCurvature<DataVector>>(background_fields);
-
-    // Compute face normal (related to the conformal metric)
+        determinant_and_inverse(spatial_metric).second;
+    const auto shift = gr::shift(spacetime_metric, inv_spatial_metric);
+    const auto lapse = gr::lapse(shift, spacetime_metric);
+
+    // Do the conformal decomposition.
+    // Note that we choose the same conformal factor as the one used for the
+    // Schwarzschild solution in isotropic coordinates. Other conformal factors
+    // could also work.
+    const auto solution_conformal_factor = solution.variables(inertial_coords,
```

Comment:
Shouldn't these be the barred coords?

Reply:
I don't think so. This is where we start to do the conformal decomposition. We could choose any conformal factor, but I chose to keep the same one as the (unboosted) Schwarzschild solution. I believe the Lorentz boost is only going to stretch spacetime, not curve it. Then, the previous conformal factor should still work.

Perhaps a better conformal decomposition would be to make the conformal metric flat (like in isotropic coordinates) and then find the proper conformal factor. Here we were just interested in testing a boosted solution with a non-unity conformal factor, so I thought it was good enough to keep the previous one. At least this means that we already know the conformal factor and that it falls off as 1/r.

Reply:
I believe evaluating the solution at the unbarred coords will fail if they are close to the horizon, which in the unbarred frame is Lorentz-contracted. Schwarzschild in isotropic coords is not horizon-penetrating, so evaluating it at coords within the horizon is problematic. You currently only use values at the outer boundary, so it's probably fine at the moment, but breaks if you try a volume integral (that's my guess).

Reply:
Hmm, that's a good point. I had not realized this. Since we can choose any conformal factor here, let's go with the one that avoids this issue. I've changed it to the barred coordinates.